### PR TITLE
Update `Code` moduledoc to mention relevant erlang functions

### PR DESCRIPTION
**Pull request by marick**
See https://elixirforum.com/t/how-does-phoenix-codereloader-work-im-trying-to-do-something-simpler/39309/5?u=marick

Note: I couldn't make `make docs` work, so I haven't actually seen the finished result. After compiling a local copy of ex_doc, I see this:

```
611 $ make docs
==> ex_doc (elixir)
CANONICAL=master/ bin/elixir ../ex_doc/bin/ex_doc "Elixir" "1.13.0-dev" "lib/elixir/ebin" --main "Kernel" --source-url "https://github.com/elixir-lang/elixir" --source-ref "be05d982d29df8192ca1c3a00990ae6f0daafbc2"  --output doc/elixir --canonical "https://hexdocs.pm/elixir/master/" --homepage-url "https://elixir-lang.org/docs.html" --formatter "html" --config "lib/elixir/docs.exs"
** (FunctionClauseError) no function clause matching in Enumerable.Range.slice/1    
    
    The following arguments were given to Enumerable.Range.slice/1:
    
        # 1
        %Inspect.Error{message: "got FunctionClauseError with message \"no function clause matching in Inspect.Range.inspect/2\" while inspecting %{__struct__: Range, first: 48, last: 57}"}
    
    Attempted function clauses (showing 1 out of 1):
    
        def slice(first.._//step = range)
```

### CODE REVIEW

**Comment by wojtekmach** [comment]
Could you update to latest ExDoc? I believe we have the Range fix on the master branch.

**Comment by wojtekmach** [comment]
Or maybe it was an Elixir bug, Id `make clean ` your elixir checkout as well as `rm -rf _build` your ExDoc one.

File: lib/elixir/lib/code.ex [hunk 1]
```diff
@@ -30,6 +30,11 @@ defmodule Code do
   file, without tracking. `eval_file/2` should be used when you are interested in
   the result of evaluating the file rather than the modules it defines.
 
+  The three functions above work with Elixir source. If you want to
+  load a compiled file (typically found below the _build directory of
+  a Mix project, with the extension .beam), see the Erlang functions
+  :code.load_file/1 and :code.load_binary/3.
```
Comment (josevalim) [inline lib/elixir/lib/code.ex:36]:
```suggestion
  The functions above work with Elixir source. If you want to load a
  module compiled to bytecode, which has the `.beam` extension
  and typically found below the _build directory of a Mix project,
  see the functions in Erlang's [`:code`](`:code`) module.
```

**Comment by josevalim** [comment]
> Note: I couldn't make make docs work, so I haven't actually seen the finished result. After compiling a local copy of ex_doc, I see this:

Removing `_build` and running `bin/elixir -S bin/mix compile` where you are using your cloned Elixir version should fix it.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**